src/cli/cli_coap_secure.cpp

```diff
@@ -0,0 +1,610 @@
+        break;
+    case OT_COAP_CODE_POST:
+        mInterpreter.mServer->OutputFormat(" (Coap Code POST)");
+        break;
+    case OT_COAP_CODE_PUT:
+        mInterpreter.mServer->OutputFormat(" (Coap Code PUT)");
+        break;
+    case OT_COAP_CODE_DELETE:
+        mInterpreter.mServer->OutputFormat(" (Coap Code DELETE)");
+        break;
+    case OT_COAP_CODE_NOT_FOUND:
+        mInterpreter.mServer->OutputFormat(" (Coap Code Not Found)");
+        break;
+    case OT_COAP_CODE_UNSUPPORTED_FORMAT:
+        mInterpreter.mServer->OutputFormat(" (Coap Code Unsupported Format)");
+        break;
+    default:
+        break;
+    }
+    mInterpreter.mServer->OutputFormat("\r\n");
+}
+
+void CoapSecureCli::PrintPayload(otMessage *aMessage) const
+{
+    uint8_t  buf[kMaxBufferSize];
+    uint16_t bytesToPrint;
+    uint16_t bytesPrinted = 0;
+    uint16_t length       = otMessageGetLength(aMessage) - otMessageGetOffset(aMessage);
+
+    if (length > 0)
+    {
+        mInterpreter.mServer->OutputFormat("    With payload [UTF8]:\r\n", aMessage);
+
+        while (length > 0)
+        {
+            bytesToPrint = (length < sizeof(buf)) ? length : sizeof(buf);
+            otMessageRead(aMessage, otMessageGetOffset(aMessage) + bytesPrinted, buf, bytesToPrint);
+
+            for (int i = 0; i < bytesToPrint; i++)
+            {
+                mInterpreter.mServer->OutputFormat("%c", buf[i]);
+            }
+            mInterpreter.mServer->OutputFormat("\r\n");
+
+            length -= bytesToPrint;
+            bytesPrinted += bytesToPrint;
+        }
+    }
+    else
+    {
+        mInterpreter.mServer->OutputFormat("    No payload.");
+    }
+
+    mInterpreter.mServer->OutputFormat("\r\n> ");
+}
+
+otError CoapSecureCli::Process(int argc, char *argv[])
+{
+    otError      error = OT_ERROR_NONE;
+    otIp6Address coapDestinationIp;
+    bool         mVerifyPeerCert = true;
+
+    VerifyOrExit(argc > 0, error = OT_ERROR_INVALID_ARGS);
+
+    if (strcmp(argv[0], "start") == 0)
+    {
+        if (argc > 1)
+        {
+            if (strcmp(argv[1], "false") == 0)
+            {
+                mVerifyPeerCert = false;
+            }
+        }
+        otCoapSecureSetSslAuthMode(mInterpreter.mInstance, mVerifyPeerCert);
+        otCoapSecureSetClientConnectedCallback(mInterpreter.mInstance, &CoapSecureCli::HandleClientConnect, this);
+        SuccessOrExit(error = otCoapSecureStart(mInterpreter.mInstance, OT_DEFAULT_COAP_SECURE_PORT, this));
+        mInterpreter.mServer->OutputFormat("Verify Peer Certificate: %s. Coap Secure service started: ",
+                                           mVerifyPeerCert ? "true" : "false");
+    }
+    else if (strcmp(argv[0], "set") == 0)
+    {
+        if (argc > 1)
+        {
+            if (strcmp(argv[1], "psk") == 0)
+            {
+                if (argc > 2)
+                {
+                    mPskLength   = strlen(argv[2]);
+                    mPskIdLength = strlen(argv[3]);
```

Comment:
Add explicit cast to address the following compiler warning:
```
src/cli/cli_coap_secure.cpp:198:36: error: implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to
      'uint8_t' (aka 'unsigned char') [-Werror,-Wconversion]
                    mPskIdLength = strlen(argv[3]);
```

Reply:
done